### ISSUE TITLE
[HOLD for payment 2024-10-24] [$250] Track - User appears with offline avatar on Submit expense page for distance request

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#48476]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.28-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4913193&group_by=cases:section_id&group_id=309130&group_order=asc
**Email or phone of affected tester (no customers):** gocemate+a2036@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to self DM.
2. Click + > Track expense > Distance.
3. From actionable whisper select Submit it to someone> Add user's email address
4. Take a look at user's avatar photo

## Expected Result:
User should be displayed with default/custom avatar photo

## Actual Result:
User appears with offline avatar on Submit expense page for distance request

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/04633c5b-d1fe-4939-bc36-ca6a90396e08

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021832021050151052347</li>
        <li>Upwork Job ID: 1832021050151052347</li>
        <li>Last Price Increase: 2024-10-01</li>
<li>Automatic offers: </li>
<ul>
<li>FitseTLT | Contributor | 104300205</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @zanyrenney</details>

**Comment by lanitochka17** [comment]
@zanyrenney  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Track - User appears with offline avatar on Submit expense page for distance request

### What is the root cause of that problem?
We are passing `includeP2P` false here
https://github.com/Expensify/App/blob/e60b8210a1f5a2471d8aa9d84d0e461774b791d6/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L113
so the user is only being added as a new invitee, hence the offline avatar.
### What changes do you think we should make in order to solve the problem?

Because we are allowing P2P distance request when it is submitted from tracked expense we need to include P2P for action `SUBMIT` so change it to 
```
            (canUseP2PDistanceRequests || action === CONST.IOU.ACTION.SUBMIT || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,

```
### What alternative solutions did you explore? (Optional)

**Comment by zanyrenney** [comment]
adding external to get a contributor to take a look at the propsals.

**Comment by abdulrahuman5196** [comment]
Checking now

**Comment by FitseTLT** [comment]
[Updated](https://github.com/Expensify/App/issues/48476#issuecomment-2327417847) to add alternative approach

**Comment by abdulrahuman5196** [comment]
@zanyrenney What is the expectation here? I am only getting `no results found` on searching anything. Should we disable the `Submit to someone` for distance requests as contributors suggestion? Or something else is the expectation?


https://github.com/user-attachments/assets/0f6319f9-264a-4f7d-be4f-801aef585e20

**Comment by FitseTLT** [comment]
@abdulrahuman5196 @zanyrenney BTW the current behaviour has made the fix even more mandatory. Now if you don't enable beta the submit to someone will navigate to the participants page but the user cannot input anything even an email/login which will confuse users in production (as beta is not enabled), the participants list will only appear if you enable beta.
So we need to hide the submit to someone button when p2p is off

https://github.com/user-attachments/assets/fba52300-6292-4fe9-b85d-211077643d45

**Comment by abdulrahuman5196** [comment]
> @abdulrahuman5196 @zanyrenney BTW the current behaviour has made the fix even more mandatory. Now if you don't enable beta the submit to someone will navigate to the participants page but the user cannot input anything even an email/login which will confuse users in production (as beta is not enabled), the participants list will only appear if you enable beta. So we need to hide the submit to someone button when p2p is off
> 2024-09-20.14-27-31.mp4

@zanyrenney I also agree on this. Seems it is not the expected behaviour, we should fix this.

**Comment by zanyrenney** [comment]
Thanks for the challenge here - that makes sense to me! Thanks for the ping.

**Comment by abdulrahuman5196** [comment]
@zanyrenney Should we block the `Submit it to Someone` flow? Or some other way we should fix it here?

**Comment by FitseTLT** [comment]
> @zanyrenney Should we block the `Submit it to Someone` flow? Or some other way we should fix it here?

@abdulrahuman5196 I think the expected behavior is obvious. `Submit to someone` for distance requests is an option to create p2p distance request which is under beta, which is why we are checking for `canUseP2PDistanceRequests` in the code base now. So we cannot allow the user to `Submit to someone` a distance request unless on beta ,`canUseP2PDistanceRequests` is enabled.

**Comment by zanyrenney** [comment]
^ this sounds good to me.

**Comment by abdulrahuman5196** [comment]
Will check today.

**Comment by MariaHCD** [comment]
Not overdue, proposal was just selected. Awaiting the PR.

**Comment by FitseTLT** [comment]
@abdulrahuman5196 PR ready

**Comment by zanyrenney** [comment]
**payment summary**

@FitseTLT requires payment [automatic offer](https://www.upwork.com/nx/wm/offer/104300205) (Contributor) - paid $250 on upwork
@abdulrahuman5196 requires payment through NewDot Manual Requests - please request $250 on ND.

**Comment by zanyrenney** [comment]
We don't need a regression test here.

**Comment by zanyrenney** [comment]
>Why shouldn't this one get a regression test?
I thought it was spotted in a testrail session from the OP, so it is already covered in TestRail.

https://expensify.testrail.io/index.php?/suites/view/14084&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=319503

Is that incorrect? @tgolen

**Comment by zanyrenney** [comment]
Definitely can update the checklist though! @abdulrahuman5196 please can you add the answers [on this](https://github.com/Expensify/App/issues/48476#issuecomment-2418319623)

**Comment by zanyrenney** [comment]
bump @abdulrahuman5196 please complete the checklist

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression, fix part of new implementation

>Determine if we should create a regression test for this bug.

Yes.

> If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


1) Go to self DM.
2) Click + > Track expense > Distance.
3) Verify that the actionable whisper of the expense created in (2) has no Submit it to someone option/button (unless you enable P2P Beta)

**Comment by zanyrenney** [comment]
Hey @abdulrahuman5196 mind doing this in the correct format from the checklist please?

**Comment by zanyrenney** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@abdulrahuman5196] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@abdulrahuman5196] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@abdulrahuman5196] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@abdulrahuman5196] Determine if we should create a regression test for this bug.
- [ ] [@abdulrahuman5196] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@zanyrenney] Link the GH issue for creating/updating the regression test once above steps have been agreed upon: https://github.com/Expensify/App/issues/51469

**Comment by abdulrahuman5196** [comment]
Hi, Updated the checklist.

**Comment by zanyrenney** [comment]
Complete!

**Comment by JmillsExpensify** [comment]
$250 approved for @abdulrahuman5196
